.github/workflows/apps/new_latest_major_versions.go

```diff
@@ -0,0 +1,307 @@
+	}
+
+	// Convert the numeric part to integers
+	num1, _ := strconv.Atoi(match1[1])
+	num2, _ := strconv.Atoi(match2[1])
+
+	if num1 <= num2 {
+		return false
+	}
+	return true
+}
+
+func main() {
+	log.SetFlags(0) // disable date and time logging
+	// Find latest major
+	github_latests := map[string]string{}  // map module (base name) => latest on github
+	contrib_latests := map[string]string{} // map module (base name) => latest on go.mod
+
+	for pkg, repository := range instrumentation.GetPackages() {
+
+		// Step 1: get the version from the module go.mod
+		fmt.Printf("package: %v\n", pkg)
+		// fmt.Printf("repository: %v\n", repository)
+
+		// if it is part of the standard packages, continue
+		if _, ok := stdLibPackages[repository]; ok {
+			continue
+		}
+
+		base := truncateVersion(string(pkg))
+		version, err := getGoModVersion(repository, string(pkg))
+		if err != nil {
+			fmt.Printf("%v go.mod not found.", pkg)
+			continue
+		}
+		fmt.Printf("version: %v\n", version)
+
+		// check if need to update contrib_latests
+		version_major_contrib := truncateMajorVersion(version)
+		if err != nil {
+			fmt.Println(err)
+			continue
+		}
+
+		if current_latest, ok := contrib_latests[base]; ok {
+			if compareVersions(version_major_contrib, current_latest) {
+				contrib_latests[base] = version_major_contrib
+			}
+		} else {
+			contrib_latests[base] = version_major_contrib
+		}
+
+		// Step 2:
+		// create the string repository@{version} and run command go list -m -json <repository>@<version>
+		// this should return a JSON
+		// extract Origin[URL] if the JSON contains it, otherwise continue
+
+		origin, err := getModuleOrigin(repository, version)
+		if err != nil {
+			fmt.Printf("Error: %v\n", err)
+			continue
+		} else {
+			fmt.Printf("origin URL: %s\n", origin)
+		}
+
+		// 2. From the VCS url, do `git ls-remote --tags <vcs_url>` to get the tags
+		// output:
+		// 3. Parse the tags, and extract all the majors from them (ex v2, v3, v4)
+		tags, err := getTags(origin)
+		if err != nil {
+			fmt.Println("Error fetching tags:", err)
+			continue
+		}
+
+		majors := groupByMajor(tags)
+
+		// 4. Get the latest version of each major. For each latest version of each major:
+		// curl https://raw.githubusercontent.com/<module>/refs/tags/v4.18.3/go.mod
+		// 5a. If request returns 404, module is not a go module. This means version belongs to the module without any /v at the end.
+		// 5b. If request returns a `go.mod`, parse the modfile and extract the mod name
+
+		// Get the latest version for each major
+		for major, versions := range majors {
+			latest := getLatestVersion(versions)
+			fmt.Printf("Latest version for %s: %s\n", major, latest)
+
+			// Fetch `go.mod` with command
+			// curl https://raw.githubusercontent.com/<module>/refs/tags/<latest>/go.mod
+			goModURL := fmt.Sprintf("https://raw.githubusercontent.com/%s/refs/tags/%s/go.mod", base, latest)
```

Comment:
I was just testing this out with other files

It looks like `https://raw.githubusercontent.com/OWNER/REPO/TAG/FILE` also works

Just wondering if there is any difference, not really seeing anything online though for it